activate

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash
+#
+# Activate the Spack environment for developing NESO and set some
+# environment variables. This script should be sourced to activate
+# the environment. You can then run the "deactivate" function to
+# return to your previous setup.
+#
+
+NESO_DEV_DIR=$(realpath $(dirname "$BASH_SOURCE"))
+
+deactivate() {
+  # Relink build directories
+  setup
+  # End the monitoring process and any `watch` processes it launched
+  if [[ ! -z ${__MONITOR_PROCESS} ]]
+  then
+      kill $__MONITOR_PROCESS
+  fi
+  ps -ef | grep "spack find -l neso nektar" | grep -v grep | awk '{print $2}' | xargs kill -9
+
+  # Tidy up environment variables
+  export I_MPI_FABRICS=$__ORIGINAL_I_MPI_FABRICS
+  export SYCL_DEVICE_FILTER=$__ORIGINAL_SYCL_DEVICE_FILTER
+  unset $__ORIGINAL_I_MPI_FABRICS
+  unset $__ORIGINAL_SYCL_DEVICE_FILTER
+  unset $__MONITOR_PROCESS
+  unset $NESO_DEV_DIR
```

Comment:
Hi @cmacmackin,
Apologies for the unsolicited review; I just tried out the new workflow and found bash complains when running deactivate; think this should be 
```suggestion
  unset __MONITOR_PROCESS
  unset NESO_DEV_DIR
```
Possibly the same for __ORIGINAL_I_MPI_FABRICS and __ORIGINAL_SYCL_DEVICE_FILTER too ?

Reply:
Yes, you are correct. I've fixed this now.